packages/rspack/src/config/defaults.ts

```diff
@@ -420,9 +424,10 @@ const applyOutputDefaults = (
 		"publicPath",
 		tp && (tp.document || tp.importScripts) ? "auto" : ""
 	);
-	D(output, "hashFunction", "xxhash64");
+
+	D(output, "hashFunction", futureDefaults ? "xxhash64" : "md4");
 	D(output, "hashDigest", "hex");
-	D(output, "hashDigestLength", 16);
+	D(output, "hashDigestLength", futureDefaults ? 16 : 20);
```

Comment:
https://github.com/webpack/webpack/blob/1e18b1d1c74e26a84e9adee519c045e2d8ed40a4/lib/config/defaults.js#L921-L923, please refer to webpack